lazyflow/operators/ioOperators/opTiffReader.py

```diff
@@ -33,97 +39,63 @@ def __init__(self, *args, **kwargs):
-
-            (P, D, Y, X, S) = first_page.shaped
-            assert P == 1, "Don't know how to handle any number of planar samples per pixel except 1 (per page)"
-            assert D == 1, "Don't know how to handle any image depth except 1"
-
-            if S == 1:
-                self._page_shape = (Y, X)
-                self._page_axes = "yx"
-            else:
-                assert shape[-3:] == (Y, X, S)
-                self._page_shape = (Y, X, S)
-                self._page_axes = "yxc"
-                assert "C" not in axes, (
-                    "If channels are in separate pages, then each page can't have multiple channels itself.\n"
-                    "(Don't know how to weave multi-channel pages together.)"
-                )
-
-            self._non_page_shape = shape[: -len(self._page_shape)]
-            assert shape == self._non_page_shape + self._page_shape
-            assert self._non_page_shape or len(pages) == 1
+            dtype_code = series.dtype
 
-            axes = axes.lower().replace("s", "c")
+            # we treat "sample" axis as "channel"
+            axes = axes.replace("s", "c")
             if "i" in axes:
                 for k in "tzc":
                     if k not in axes:
                         axes = axes.replace("i", k)
                         break
                 if "i" in axes:
-                    raise RuntimeError(
-                        "Image has an 'I' axis, and I don't know what it represents. "
-                        "(Separate T,Z,C axes already exist.)"
+                    raise UnsupportedTiffError(
+                        filepath=self._filepath,
+                        details=f"Image has an 'I' axis, and I don't know what it represents (separate T,Z,C axes already exist): {axes}",
                     )
 
+            # tifffile will add potentially multiple "q" axes to data when saving without specifying them
             if "q" in axes:
-                old_axes = axes
                 axes = get_default_axisordering(shape)
-                logger.warning(
-                    f"Unknown axistags detected - assuming default axis order. Guessed {axes} from {old_axes}."
+
+            if any(ax not in "tzyxc" for ax in axes) or len(shape) > 5:
+                raise UnsupportedTiffError(
+                    filepath=self._filepath,
+                    details=f"Don't know how to read TIFF files with more than 5 dimensions (Your image has {len(shape)} dimensions). Axes: {axes}.",
                 )
 
-            self.Output.meta.shape = shape
-            self.Output.meta.axistags = vigra.defaultAxistags(str(axes))
-            self.Output.meta.dtype = numpy.dtype(dtype_code).type
-            self.Output.meta.ideal_blockshape = ((1,) * len(self._non_page_shape)) + self._page_shape
+            self._page_axes = tiff_file.series[0].pages[0].axes.lower()
+            self._page_shape = tiff_file.series[0].pages[0].shape
```

Comment:
```suggestion
            self._page_axes = series.pages[0].axes.lower()
            self._page_shape = series.pages[0].shape
```

Reply:
Follow-up: are we sure that all TIFF pages always have the same shape/axes?

Reply:
https://github.com/cgohlke/tifffile/blob/ba27d94def41216bc024b2629659d72a478f0c49/tifffile/tifffile.py#L11681-L11682